mplc/doc/documentation.md

```diff
@@ -0,0 +1,497 @@
+  Example: `init_model_from='./my_previously_trained_models/weights_super_model.h5'`
+
+- `dataset_proportion`: `float` (default: `1`).  
+  This argument allows you to make computation on a sub-dataset of the provided dataset.
+  This is the proportion of the dataset (initially the train and test sets) which is randomly selected to create a sub-dataset,
+  it's done before the creation of the global validation set.
+  You have to ensure that `0 < dataset_proportion <= 1`.  
+  Example: `dataset_proportion=0.2`
+
+### Definition of collaborative scenarios
+
+- `partners_count`: `int`  
+  Number of partners in the mocked collaborative ML scenario.  
+  Example: `partners_count=4`
+
+- `amounts_per_partner`: `[float]`  
+  Fractions of the original dataset each partner receives to mock a collaborative ML scenario where each partner provides data for the ML training.  
+  You have to ensure the fractions sum up to `1`.
+  Example: `amounts_per_partner=[0.3, 0.3, 0.1, 0.3]`
+
+<a id="sample_split_option"></a>
+- `samples_split_option`: `['basic', 'random']` (default), `['basic', 'stratified']` or `['advanced', [[nb of clusters (int), 'shared' or 'specific']]]`   
+  How the original dataset data samples are split among partners:
+  
+    - `'basic'` approaches:
+      - `'random'`: the dataset is shuffled and partners receive data samples selected randomly
+      - `'stratified'`: the dataset is stratified per class and each partner receives certain classes only (note: depending on the `amounts_per_partner` specified, there might be small overlaps of classes)
+    - `'advanced'` approach `[[nb of clusters (int), 'shared' or 'specific']]`: in certain cases it might be interesting to split the dataset among partners in a more elaborate way. For that we consider the data samples from the initial dataset as split in clusters per data labels. The advanced split is configured by indicating, for each partner in sequence, the following 2 elements:
+      - `nb of clusters (int)`: the given partner will receive data samples from that many different clusters (clusters of data samples per labels/classes)
+      - `'shared'` or `'specific'`:
+        - `'shared'`: all partners with option `'shared'` receive data samples picked from clusters they all share data samples from
+        - `'specific'`: each partner with option `'specific'` receives data samples picked from cluster(s) it is the only one to receive from
+
+  Example: `samples_split_option=['advanced', [[7, 'shared'], [6, 'shared'], [2, 'specific'], [1, 'specific']]]`
+
+![Example of the advanced split option](../../img/advanced_split_example.png)
+
+- `corrupted_datasets`: `[not_corrupted (default), shuffled or corrupted]`  
+  Enables to artificially corrupt the data of one or several partners:
+
+    - `not_corrupted`: data are not corrupted
+    - `shuffled`: labels are shuffled randomly, not corresponding anymore with inputs
+    - `corrupted`: labels are all offset of `1` class
```

Comment:
It seems stange... It is not exactly what is implemented. 
Shuffle does not shuffle the labels between the data, but randomly flip a label to another. It is a major difference, as if a partner's dataset contains only labels 1, 2 and 3, after the shuffling, his labels can be 0, 1, 2 , 3...,8 or  9.  

The corrupted label are not all offset to `1` class, but permuted. `1`s become `2`s,  `2`s become `3`s,  `3`s become `4`s, and so on. Practically, we apply a Circulant matrix

Reply:
- shuffled: OK, I missed this change
- corrupted: this is what is meant already, I'll make it clearer